lib/Slack/RTM/Bot/Client.pm

```diff
@@ -155,13 +155,23 @@ sub _refetch_conversation_name {
 
 sub _refetch_conversations {
 	my $self = shift;
-	my $res = $ua->request(GET "https://slack.com/api/conversations.list?limit=999999999&types=public_channel,private_channel,mpim,im&token=$self->{token}");
+	my $res;
 	eval {
-		$self->{info}->{channels} = Slack::RTM::Bot::Information::_parse_conversations(JSON::from_json($res->content));
-	};
-	if ($@) {
-		die '_refetch_conversations response fail:'.Dumper $res->content;
-	}
+		my $conversations = {};
+		my $cursor = "";
+		do {
+			$res = $ua->request(GET "https://slack.com/api/conversations.list?types=public_channel,private_channel&token=$self->{token}&cursor=$cursor&limit=1000");
+			my $args = JSON::from_json($res->content);
+			for my $conversation (@{$args->{channels}}) {

```

Comment:
Can't use `Slack::RTM::Bot::Information::_parse_conversations` ?
I want to avoid code duplication.

```perl
$conversations = {%$conversations, %{Slack::RTM::Bot::Information::_parse_conversations($args)}};
```

Reply:
_parse_conversations (and _parse_users) seem like unnecessary functions for a tiny 'for' loop. Do you know of any use cases (besides tests) where users or conversations would be manually loaded?

Reply:
> parse_conversations (and _parse_users) seem like unnecessary functions for a tiny 'for' loop

I understand. 

> Do you know of any use cases (besides tests) where users or conversations would be manually loaded?

I'm sorry. I have not prepared enough tests.